test_4_filter_numbers.py

```diff
@@ -1,7 +1,8 @@
 def filter_numbers(arr):
   # Remove the `return None` code below and add your own code.
-  return None
-
+  return arr < 10
+  filter(filter_numbers, arr)
```

Comment:
Second: In this line you are calling the function `filter_numbers` within its own definition.